sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -321,6 +321,27 @@ public static String toTableSpec(TableReference ref) {
     return sb.toString();
   }
 
+  private static String tableProvider(ValueProvider<String> table) {
```

Comment:
Why is this called `tableProvider`? Isn't the `ValueProvider<String>` the table provider? Should it be called something like `tableOrProviderString`? If not, is that at least conceptually correct (I haven't looked at the providers extensively)?